src/nvim/api/private/helpers.c

```diff
@@ -2095,3 +2097,20 @@ bool parse_float_config(Dictionary config, FloatConfig *fconfig, bool reconf,
   }
   return true;
 }
+
+/// Sets sctx for API calls.
+///
+/// @param channel_id     api clients id. Used to determine if it's a internal
+///                       call or a rpc call.
+/// @return returns       previous value of current_sctx. To be used
+///                       to be used for restoreing sctx to previos state.
+sctx_T api_set_sctx(uint64_t channel_id)
+{
+  sctx_T old_current_sctx = current_sctx;
+  if (channel_id != VIML_INTERNAL_CALL) {
+    current_sctx.sc_sid =
+      channel_id == LUA_INTERNAL_CALL ? SID_LUA : SID_API_CLIENT;
+    current_sctx.sc_lnum = 0;
```

Comment:
todo: we really need to enhance :verbose for API calls. `api_set_sctx(..., fn_name, code)` could optionally take a `fn_name` and `code` which will say the API method name, and optionally some snippet of code which gives a hint. Also sctx should store the API channel id, so that `:verbose` can report the channel info (client name at least).